### PR TITLE
feat: Filtering and pagination on frontend

### PR DESCRIPTION
**Pull request by carrymisss**
Made the logic for pagination and filtering, made components and integrated into the application.

### CODE REVIEW

File: mobile/components/Todos/TodosList.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                                                setSearchParam={setSearchParam}
+                                                setCompletionParam={setCompletionParam}
+                                                setPrivacyParam={setPrivacyParam}
+                                            />
+                                        </Flex>
+                                        <WhiteSpace size="lg" />
+                                    </View>
+                                </WingBlank>
+                            }
+                            renderItem={
```
Comment (CodeGeneration-2020) [inline mobile/components/Todos/TodosList.tsx:80]: move it to a separate function

File: mobile/components/Todos/TodosList.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                    <Text style={styles.emptyText}>Error</Text>
+                </View> :
+                    status === "loading" ?
+                    <View>
+                        <WhiteSpace size="lg" />
+                        <ActivityIndicator size="large" color={PRIMARY_COLOR} />
+                    </View> :
+                        <FlatList
+                            data={data}
+                            ListHeaderComponent={
```
Comment (CodeGeneration-2020) [inline mobile/components/Todos/TodosList.tsx:60]: move this component to a separate file (module) in order to reduce lines of code for this file. Everything should be consolidated across modules

File: mobile/containers/Todos/ParamsFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        if (enableCompletionParam) {
+            setCompletionParam(undefined);
+        } else {
+            setCompletionParam(true);
+        }
+    };
+
+    const handleEnabledPrivacyParam = (): void => {
+        setEnablePrivacyParam(!enablePrivacyParam);
+        if (enablePrivacyParam) {
```
Comment (CodeGeneration-2020) [inline mobile/containers/Todos/ParamsFilter.tsx:32]: this logic is weird
`setPrivacyParam(enablePrivacyParam ? undefined : 'public')`
if / else is redundant here

File: mobile/containers/Todos/ParamsFilter.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        queryClient.fetchQuery(REACT_QUERY_KEYS.ALL_TODOS);
+    };
+
+    const handleResetFilters = (): void => {
+        setEnableCompletionParam(false);
+        setEnablePrivacyParam(false);
+        setCompletionParam(undefined);
+        setPrivacyParam(undefined);
+        setSearchParam(undefined);
+        setTimeout(() => {
```
Comment (CodeGeneration-2020) [inline mobile/containers/Todos/ParamsFilter.tsx:67]: is there a reason we have this expression with `setTimeout`?

**Review by CodeGeneration-2020 — CHANGES_REQUESTED** [review]

File: mobile/containers/Todos/ParamsFilter.tsx (hunk 4 above)
Comment (carrymisss) [reply mobile/containers/Todos/ParamsFilter.tsx:67]: Yes, without it the "Reset" doesn't work

**Pull request merged**